plugins/content-manager/src/main/java/com/wazuh/contentmanager/util/JsonPatch.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (C) 2024, Wazuh Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.wazuh.contentmanager.util;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Utility class for applying JSON Patch operations to JSON documents.
+ *
+ * <p>This class provides methods to apply various JSON Patch operations such as add, remove,
+ * replace, move, copy, and test. It uses the Gson library for JSON manipulation.
+ */
+public class JsonPatch {
+    private static final Logger log = LogManager.getLogger(JsonPatch.class);
+
+    /**
+     * Applies a single JSON Patch operation to a document.
+     *
+     * @param document The target JSON document.
+     * @param operation The JSON Patch operation.
+     */
+    public static void applyOperation(JsonObject document, JsonObject operation) {
+        String op = operation.get("op").getAsString();
+        String path = operation.get("path").getAsString();
+        JsonElement value = operation.has("value") ? operation.get("value") : null;
+        String from = operation.has("from") ? operation.get("from").getAsString() : null;
+
+        switch (op) {
+            case "add":
+                addOperation(document, path, value);
+                break;
+            case "remove":
+                removeOperation(document, path);
+                break;
+            case "replace":
+                replaceOperation(document, path, value);
+                break;
+            case "move":
+                moveOperation(document, from, path);
+                break;
+            case "copy":
+                copyOperation(document, from, path);
+                break;
+            case "test":
+                testOperation(document, path, value);
+                break;
```

Comment:
Missing `this` on all these method calls

Reply:
Applied on https://github.com/wazuh/wazuh-indexer-plugins/pull/362/commits/f21a4dcd5078bb43a5cb4168fd1d6f35bec079c0